Test.cpp

```diff
@@ -86,12 +98,12 @@ int main(int argc, char const *argv[])
   // ANS: Here it is an example of how to handle that case.
   vc::compiler_ptr_t clang = vc::make_compiler<vc::SystemCompilerOptimizer>(
                                           "llvm/clang",
-                                          "clang",
-                                          "opt",
+                                          std::filesystem::u8path(CLANG_EXE_NAME),
+                                          std::filesystem::u8path(OPT_EXE_FULLPATH),
                                           ".",
-                                          "./test.log",
-                                          "/usr/bin",
-                                          "/usr/lib/llvm/bin"
+                                          "./test.log", 
```

Comment:
can this one go to `filesystem` style as well?

Reply:
If you prefer, I'll do it but it won't change much, it will still invoke the std::filesystem constructor and it won't need the std::filesystem::u8path (which is deprecated in c++20 but required in c++17) due to the fact that it does not contain "risky" utf chars

Reply:
What I would like to achieve is a consistent format across all the project files.

If the C++ standard creates an issue, we can have an helper header containing something like

`
#if defined(__cplusplus) && __cplusplus <= 201703L

using vc::mypath = std::filesystem::u8path;

#else

using vc::mypath = std::string;

#endif
`

Reply:
Ok, I had replaced them all with std::filesystem::u8path!